### PR TITLE
Fix4967 i mvc core builder add json options

### PR DESCRIPTION
**Pull request by iscifoni**
Add IMvcCoreBuilder.AddJsonOptions extension method
#4967

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Formatters.Json/DependencyInjection/MvcJsonMvcCoreBuilderExtensions.cs [hunk 1]
```diff
@@ -47,6 +47,24 @@ public static IMvcCoreBuilder AddJsonFormatters(this IMvcCoreBuilder builder)
             return builder;
         }
 
+        public static IMvcCoreBuilder AddJsonOptions(
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Formatters.Json/DependencyInjection/MvcJsonMvcCoreBuilderExtensions.cs]: It's a public extension method for a public type - no harm adding docs for this.

File: src/Microsoft.AspNetCore.Mvc.Formatters.Json/DependencyInjection/MvcJsonMvcCoreBuilderExtensions.cs [hunk 2]
```diff
@@ -47,6 +47,12 @@ public static IMvcCoreBuilder AddJsonFormatters(this IMvcCoreBuilder builder)
             return builder;
         }
 
+        /// <summary>
+        /// Adds configuration of <see cref="MvcJsonOptions"/> for the application.
+        /// </summary>
+        /// <param name="builder">The <see cref="IMvcCoreBuilder"/>.</param>
+        /// <param name="setupAction">The <see cref="MvcJsonOptions"/> which need to be configured.</param>
+        /// <returns></returns>
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Formatters.Json/DependencyInjection/MvcJsonMvcCoreBuilderExtensions.cs]: `<returns>The <see cref="IMvcCoreBuilder"/>.</returns>`

File: test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs [hunk 3]
```diff
@@ -31,6 +31,8 @@
 using Microsoft.Extensions.Options;
 using Moq;
 using Xunit;
+using Newtonsoft.Json.Serialization;
+using Newtonsoft.Json;
```
Comment (javiercn) [inline test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs]: nit: Sort the using statements. System namespaces first then other namespaces in alphabetical order.

**Comment by javiercn** [comment]
:shipit: After the comments are addressed.

File: src/Microsoft.AspNetCore.Mvc.Formatters.Json/DependencyInjection/MvcJsonMvcCoreBuilderExtensions.cs (hunk 2 above)
Comment (iscifoni) [inline src/Microsoft.AspNetCore.Mvc.Formatters.Json/DependencyInjection/MvcJsonMvcCoreBuilderExtensions.cs]: ops sorry :smile:

File: test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs (hunk 3 above)
Comment (iscifoni) [inline test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs]: :+1:

File: test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs [hunk 4]
```diff
@@ -226,6 +228,22 @@ public void AddMvcCore_ReusesExistingApplicationPartManagerInstance_IfFoundOnSer
...
+           
+            services.AddMvcCore().
+                AddJsonOptions((options) =>
+            {
+                options.SerializerSettings.ContractResolver = new CamelCasePropertyNamesContractResolver();
+                options.SerializerSettings.Formatting = Formatting.Indented;
+            });
+
+            // Assert
+            var descriptor = Assert.Single(services, d => d.ServiceType == typeof(IConfigureOptions<MvcJsonOptions>));
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs]: I can make this change as part of merging this - you don't need to assign this to a variable if you don't intend on doing anything with it. I'm surprised the compiler doesn't warn here.

**Comment by pranavkm** [comment]
@iscifoni I'll get this squashed and merged in a bit. For future reference, we do all our work in the `dev` branch. The `master` branch points to our most recent release, so you almost never want that except using it as a reference.

File: test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs (hunk 4 above)
Comment (iscifoni) [inline test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs]: I have doing anoter test now,  no warning is present

**Comment by iscifoni** [comment]
Ok thanks.

**Comment by pranavkm** [comment]
Merged in https://github.com/aspnet/Mvc/commit/3d2bb673ee8b55e278db304fae8da2fba462bb9e. Thanks for the PR!

**Pull request closed without merging**